fms2_io/fms_netcdf_domain_io.F90

```diff
@@ -743,15 +743,16 @@ end subroutine domain_offsets
 
 !> @brief Get starting/ending global indices of the I/O domain for a domain decomposed
 !!        file.
-subroutine get_global_io_domain_indices(fileobj, dimname, is, ie)
+subroutine get_global_io_domain_indices(fileobj, dimname, is, ie, indices)
 
   type(FmsNetcdfDomainFile_t), intent(in) :: fileobj !< File object.
   character(len=*), intent(in) :: dimname !< Name of dimension variable.
   integer, intent(out) :: is !< Staring index of I/O global domain.
   integer, intent(out) :: ie !< Ending index of I/O global domain.
+  integer, dimension(:), allocatable, intent(inout), optional :: indices
```

Comment:
This implies that indices always has to be on the heap.  Are you sure that's what you want?  Also, I want to agree with Jess about doxygen commenting on what this variable is.

Reply:
> You really need to add the doxy comment for the `indices` variable.
> 
> I don't think deallocating and reallocating a variable sent in and changing the size of the dimension is a good idea. It should come in with the correct size IMO.

The code doesn't know the size of `indices` before  `get_global_io_domain_indices` is called. The size of `indices` is defined by `is`, `ie`, which are obtained after the ` mpp_get_global_domain` call

I could (should?) change it from `inout` to just `out`

Reply:
> I could (should?) change it from `inout` to just `out`
I think that makes a lot more sense.  It needs to be vary clear that this array **WILL** be changed.